homeassistant/components/locative/__init__.py

```diff
@@ -117,6 +110,7 @@ async def handle_webhook(hass, webhook_id, request):
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Configure based on config entry."""
+    hass.data.setdefault(DOMAIN, {"devices": set(), "unsub_device_tracker": {}})
```

Comment:
Note: I don't like that this creates a new dict (with a new set and a new inner dict) on every entry setup, even if the key is already present.
Is this not a case where an IF statement would be better?

Reply:
Also shouldn't the unsub device tracker be able to be added to the `on_unload` on the config entry.

Reply:
> I don't like that this creates a new dict (with a new set and a new inner dict) on every entry setup

I really don't think that's a valid comment, this is the pattern we generally use for creating hass data shared between several config entries; there are 400+ hits of `hass.data.setdefault(DOMAIN` across the code base.
It sucks there's no lazy one-line version of `dict.setdefault` I guess.